src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1690,19 +2021,20 @@ private void onDecodeFrameHeader(
             Consumer<Builder> mutator)
         {
             amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
-                    .size(0)
-                    .doff(frameHeader.doff())
-                    .type(frameHeader.type())
-                    .channel(frameHeader.channel())
-                    .performative(mutator);
+                .size(0)
+                .doff(frameHeader.doff())
+                .type(frameHeader.type())
+                .channel(frameHeader.channel())
+                .performative(mutator);
         }
 
         private void onDecodeOpen(
             long traceId,
             long authorization,
             AmqpOpenFW open)
         {
-            doEncodeOpen(traceId, authorization, open.hasMaxFrameSize(), (int) open.maxFrameSize());
+            this.encodeMaxFrameSize = Math.min(replySharedBudget, open.maxFrameSize());
```

Comment:
Shouldn't `replySharedBudget` be  replaced by `bufferPool.slotCapacity()` here?

Also, this is where `AmqpServer` member field `decodeMaxFrameSize` should be set to `maxFrameSize` taken from config.